engine/src/main/scala/cromwell/engine/workflow/tokens/JobTokenDispenserActor.scala

```diff
@@ -22,18 +26,35 @@ import scala.concurrent.duration._
 import scala.util.{Failure, Success, Try}
 
 class JobTokenDispenserActor(override val serviceRegistryActor: ActorRef,
-                             override val distributionRate: DynamicRateLimiter.Rate,
+                             override val dispensingRate: DynamicRateLimiter.Rate,
                              logInterval: Option[FiniteDuration],
                              dispenserType: String,
                              tokenAllocatedDescription: String)
-  extends Actor with ActorLogging with JobInstrumentation with CromwellInstrumentationScheduler with Timers with DynamicRateLimiter {
+  extends Actor
+    with ActorLogging
+    with JobInstrumentation
+    with CromwellInstrumentationScheduler
+    with Timers
+    with DynamicRateLimiter
+    with CromwellInstrumentation
+{
+
+  // Metrics paths are based on the dispenser type
+  private val tokenDispenserMetricsBasePath: NonEmptyList[String] = NonEmptyList.of("token_dispenser", dispenserType)
+
+  private val tokenLeaseDurationMetricPath: NonEmptyList[String] = tokenDispenserMetricsBasePath :+ "token_hold_duration"
+
+  private val tokenDispenserMetricsActivityRates: NonEmptyList[String] = tokenDispenserMetricsBasePath :+ "activity_rate"
+  private val requestsEnqueuedMetricPath: NonEmptyList[String] = tokenDispenserMetricsActivityRates :+ "requests_enqueued"
+  private val tokensLeasedMetricPath: NonEmptyList[String] = tokenDispenserMetricsActivityRates :+ "tokens_dispensed"
+  private val tokensReturnedMetricPath: NonEmptyList[String] = tokenDispenserMetricsActivityRates :+ "tokens_returned"
```

Comment:
Will these forever increase? If so, do we have to worry about hitting a numeric limit?

Reply:
(I probably should have also implicated `requests_enqueued` in this comment)

Reply:
Depends how you graph them. We normally graph them as rates rather than raw counts so it doesn't become a problem. And I assume we never get close enough to the max value between restarts that rollover ever becomes a problem.

Reply:
are the readers and summarizer going to be posting a bunch of zeros that throw off what's going on with the runners?

Reply:
(this is not the fastest growing metric in the system by a long way and I don't remember seeing other metrics have similar problems)

Reply:
> are the readers and summarizer going to be posting a bunch of zeros that throw off what's going on with the runners? 

Good question, but only if we aggregate across all instances. in that case it depends whether they create token dispensers which sit doing nothing, or never create them in the first place.

Reply:
It appears that they do create dispensers and emit these metrics. for the sake of expediency, I'm going to merge as-is and create a follow up ticket to not publish for non-runners